### ISSUE TITLE
Integrated Terminal Backlog

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#9959]
This is a ranked backlog of the larger, heavy hitter items within the [overall integrated terminal backlog](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aintegrated-terminal).

1. ✅ Expose integrated terminal extension API #9957
2. ✅ Add scroll bar to integrated terminal #6602
3. Integrated terminal does not launch when project folder contains special/unicode characters #7727
4. Add virtual selection to integrated terminal #9958

This list does not contain smaller issues, check the milestone the specific issue is associated with for that. The full backlog of integrated terminal issues is available [here](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aintegrated-terminal).

**Comment by whitecolor** [comment]
@Tyriar 
Is it possible to have in terminal features that exist in debug console output like ability to inspect/extend objects ala browser console? Also clickable links to code? Are they based on totally different technologies?

**Comment by Tyriar** [comment]
@whitecolor there is stuff happening around bringing the integrated terminal and debugging a little closer, that's the main use case for #9957 but we're only just exploring it right now. Clickable links is being tracked here #7321, before starting on this I think the upstream issue relating to the linkify addon needs to be looked at https://github.com/sourcelair/xterm.js/issues/27

**Comment by whitecolor** [comment]
Cool. Though that seems a lot of work)

**Comment by Tyriar** [comment]
Removed "Backspace, left and right in Windows integrated terminal cause prompt to go out of sync #7345" from the list as it only impacts git bash and is certainly an upstream issue.

**Comment by Tyriar** [comment]
Closing this off, no longer needed for tracking.
